components/infobox/extensions/commons/player_introduction.lua

```diff
@@ -0,0 +1,595 @@
+		playerInfo = self:_playerQuery()
+	end
+
+	self:_parsePlayerInfo(args, playerInfo)
+
+	if Table.isEmpty(self.playerInfo) then
+		return self
+	end
+
+	self:_readTransferData(args.transferquery)
+
+	self:_roleAdjusts(args)
+
+	self.options = {
+		showRole = Logic.readBool(args.show_role),
+		showFaction = Logic.readBool(args.show_faction),
+	}
+
+	return self
+end
+
+function PlayerIntroduction:_playerQuery()
+	local queryData = mw.ext.LiquipediaDB.lpdb('player', {
+		conditions = '[[pagename::' .. self.player .. ']]',
+		query = 'id, name, localizedname, type, nationality, nationality, nationality2, nationality3, '
+			.. 'birthdate, deathdate, status, extradata, teampagename',
+	})
+
+	if type(queryData[1]) == 'table' then
+		return queryData[1]
+	end
+
+	return {}
+end
+
+---@param args argsValues
+---@param playerInfo table
+---@return nil
+function PlayerIntroduction:_parsePlayerInfo(args, playerInfo)
+	playerInfo.extradata = playerInfo.extradata or {}
+
+	local role = (args.role or playerInfo.extradata.role or ''):lower()
+
+	local personType = Logic.emptyOr(args.type, playerInfo.type, TYPE_PLAYER):lower()
+	if personType ~= TYPE_PLAYER and String.isNotEmpty(role) then
+		personType = role
+	end
+
+	local name = args.name or playerInfo.name
+
+	local nameArray = String.isNotEmpty(name)
+		and mw.text.split(name, ' ')
+		or {}
+
+	local function readNames(prefix)
+		return Array.extractValues(Table.filter(args, function(entry) return entry:find('^' .. prefix .. '%d+$') end))
+	end
+
+	self.playerInfo = {
+		team = args.team or playerInfo.teampagename,
+		team2 = args.team2 or playerInfo.extradata.team2,
+		name = name,
+		status = Logic.emptyOr(args.status, playerInfo.status, 'active'):lower(),
+		type = personType:lower(),
+		game = Logic.emptyOr(args.game, playerInfo.extradata.game, args.defaultGame),
+		id = args.id or playerInfo.id,
+		idIPA = args.idIPA or playerInfo.extradata.idIPA,
+		idAudio = args.idAudio or playerInfo.extradata.idAudio,
+		birthDate = args.birthdate or playerInfo.birthdate,
+		deathDate = args.deathdate or playerInfo.deathdate,
+		nationality = args.nationality or playerInfo.nationality,
+		nationality2 = args.nationality2 or playerInfo.nationality2,
+		nationality3 = args.nationality3 or playerInfo.nationality3,
+		faction = args.faction or playerInfo.extradata.faction,
+		faction2 = args.faction2 or playerInfo.extradata.faction2,
+		faction3 = args.faction3 or playerInfo.extradata.faction3,
+		subText = args.subtext,
+		freeText = args.freetext,
+		role = role,
+		role2 = (args.role2 or playerInfo.extradata.role2 or ''):lower(),
+		firstName = Logic.emptyOr(args.firstname, playerInfo.extradata.firstname, table.remove(nameArray, 1)),
+		lastName = Logic.emptyOr(args.lastname, playerInfo.extradata.lastname, nameArray[#nameArray]),
+		formerlyKnownAs = readNames('formername'),
+		alsoKnownAs = readNames('aka'),
+	}
+end
+
+
+---@param queryType 'datapoint'?
```

Comment:
.

Reply:
yeah well technically any other string is treated the same as nil
so for the sake of doing it only this one is allowed
alternatively could read it in parse (and the legacy entry point for `Module:PlayerTeamAuto`) and make it a bool